docs/_posts/2023-04-20-functional-modularity-2.md

```diff
@@ -3,56 +3,80 @@ title: "Modularity With Functional Programming Part 2"
+We covered examples from the "Gluing Functions Together" portion of John Hughes's 1984 memo titled "Why Functional Programming Matters." This second post will build on those ideas and cover two more examples from that section of the memo with accompanying Scala 3 sample code.
 
-After this post, we will move on to the second idea Hughes presented in his memo, "Gluing Programs Together", which focuses on a feature common to many functional programming languages known as lazy evaluation.
+The two examples in this post extend the first post's "functions as glue" concept to show that the ideas of modularity through function composition and higher order functions can be used to perform operations on more complex data structures such as lists of lists and trees.
 
-If you missed the first post in this series, you can find it [here](/2023/03/functional-modularity-1/)), I recommend going back and reading that post as it explains the ideas these examples are expanding upon.
+Following this post the series will move on to the second idea Hughes presented in his memo, "Gluing Programs Together", which focuses on a feature common to many functional programming languages known as "lazy evaluation."
+
+If you missed the first post in this series, you can find it [here](/2023/03/functional-modularity-1/), I recommend going back and reading it if the ideas of modularity and higher order functions presented here are feeling unfamiliar.
 
 ## Example 1: Higher Order Functions To Operate On Matrices
 
-If you would like to follow along, all you need is the Scala 3 tools installed on your machine. View instructions specific to your OS [here on the scala download page](https://www.scala-lang.org/download/). While in the first post I recommended using a scala 3 REPL to code these examples, because we start working with language constructs in scala that expect us to organize our code into files, I found it easier to use a [scala 3 worksheet](https://docs.scala-lang.org/scala3/book/tools-worksheets.html) in Visual Studio Code for these examples.
+If you would like to follow along, all you need is the Scala 3 tools installed on your machine. View instructions specific to your OS [here on the scala download page](<https://www.scala-lang.org/download/>).
+
+While in the first post I recommended using a scala 3 REPL to code these examples, for these I recommend an editor like Visual Studio Code and [scala 3 worksheets](<https://docs.scala-lang.org/scala3/book/tools-worksheets.html>) instead. We start working with slightly larger programs in these examples, and it will be easier to write longer procedures out and save them for later instead of submitting commands to a CLI interpreter. With worksheets we still get the benefit of not having to perform manual compile and run steps each time we want to see intermediate results.
```

Comment:
```suggestion
While in the first post I recommended using a scala 3 REPL to code these examples, for these I recommend an editor like Visual Studio Code and [scala 3 worksheets](https://docs.scala-lang.org/scala3/book/tools-worksheets.html) instead. We start working with slightly larger programs in these examples, and it will be easier to write longer procedures out and save them for later instead of submitting commands to a CLI interpreter. With worksheets we still get the benefit of not having to perform manual compile and run steps each time we want to see intermediate results.
```